Intersections_3/include/CGAL/Intersections_3/internal/Bbox_3_Sphere_3_do_intersect.h

```diff
@@ -35,39 +35,59 @@ namespace internal {
         typedef typename K::Point_3 Point;
         FT d = FT(0);
         FT distance = FT(0);
+        FT sr = sphere.squared_radius();
+
                 Point center = sphere.center();
 
                 if(center.x() < (FT)bbox.xmin())
                 {
                         d = (FT)bbox.xmin() - center.x();
-                        distance += d * d;
+                        d = square(d);
+                        if(d > sr){
+                          return false;
+                        }
```

Comment:
I do not think that code can be trusted. `d` is a construction, and its comparison to `sr` is not exact. It can be wrong in both way (false-positive or false-negative).

Reply:
Actually, the issue is elsewhere: If `FT` is `Interval_nt`, then this comparison could throw the exception `Uncertain_conversion_exception` too early. There should be a `certainly`:
```suggestion
                        d = square(d);
                        if(certainly(d > sr)){
                          return false;
                        }
```
and the same for all the comparisons supposed to trigger an early `return false`.

Reply:
it is exact because we are using filtered predicates (i.e. interval or gmpq)